### ISSUE TITLE
Source tweaks

### ISSUE THREAD

**Issue by drewbanin** [fishtown-analytics/dbt#1267]
### Feature description
[Sources](https://github.com/fishtown-analytics/dbt/pull/1254) are going to be a great feature in dbt v0.13.0! There are a couple of tiny tweaks we should make to ensure that we're covering some expected use cases, both involving package management.

The original sources spec was very deliberate about using `sql_table_name` to point a `source` to a table in the database. While this works, it also falls short in two areas:

1. configuring sources inside of packages
2. making `sources` consistent with Relations

Instead, sources should be configured with a `database`, `schema`, and `identifier` as shown below.

**configuring sources inside of packages*

Example:
```
sources:
  - name: snowplow
    loader: snowplow
    database: raw # Optional, defaults to None
    schema: atomic # Optional, defaults to `name` config

    tables:
        - name: events
          identifier: events # Optional, defaults to `name` config
```

This strikes a balance between:
1) supporting unconventionally named source tables (like `RaW.12snowplow.event-stream`, which the user can supply as quoted values for db/schema/table)
2) reducing the need to repeat yourself. Sources will largely map 1-to-1 with schemas, and it therefore makes sense to only specify the database and schema only once.

In packages specifically, we're in the habit of typing out a full database/schema/table name for each source table as shown here: https://github.com/fishtown-analytics/quickbooks/blob/master/dbt_project.yml#L23

Instead, these packages should accept as variables a database and a schema. From there, the package author can determine if further variables are required for specific tables, or if the table names will be consistent across deployments of the package.

N.B.: I'm not 100% sold on these identifiers yet, and it will be good to use dbt's config aliasing functionality to substitute `schema` with `dataset` on BQ (or similar).

**making `sources` consistent with Relations**

At present, the `ref` function returns a `Relation` object. This is convenient, because Relations are the primary currency of dbt's adapter API. There are myriad use cases for called adapter functions on source tables (like `get_columns_in_relation`). The `sql_table_name` config makes this really difficult to do effectively. Instead, supplying a separate database, schema, and table for each source will dbt to 1) generate and 2) return a `Relation` when the `source()` function is used.

## Future work: extending sources

Let's not do anything on this front for now, but at some point, it will be useful to be able to extend sources with configs from a parent project. This will typically take the shape of overriding descriptions (or freshness configs) for a package. We'll need to figure out a good syntax for this, and we'll also need to wrangle in a notion of namespacing to do it well. No action required, but wanted to plant the seed :)

**Comment by beckjake** [comment]
Fixed in #1278
